src/main/kotlin/edu/byu/mealplanningassistant/services/MealPlanGenerator.kt

```diff
@@ -0,0 +1,22 @@
+package edu.byu.mealplanningassistant.services
+
+import edu.byu.mealplanningassistant.models.GetRandomizedRecipesRequest
+import edu.byu.mealplanningassistant.models.GetRandomizedRecipesResponse
+
+class MealPlanGenerator {
+    fun getRandomizedRecipes(getRandomizedRecipesRequest: GetRandomizedRecipesRequest) : GetRandomizedRecipesResponse {
+        TODO()
+        /**
+         * queryRecordsInRandomOrder().forEach {
+         *      if it.matches(getRandomizedRecipeRequests.requests.any()) {
```

Comment:
This is pseudo code inspired by Kotlin, so not exactly, but I'll explain what parts of it are actually Kotlin.

`queryRecordsInRandomOrder()` is a function we would have to write. If it returns an iterable data structure (eg. a `List`). Every Iterable data structure has a `forEach` method that takes a lambda function (if the last argument of a function is a lambda expression, the lambda expression does not need to go inside of the parens and if it's the only argument, it doesn't even need parens. This avoids having lambda functions wrapper in `({})` and instead it's just `{}`). Inside of a lambda function that is passed an object, the object has a default name of `it`. This can be changed to whatever you want by putting `<nameOfVar> ->` at the top of the lambda function. The if statement would technically need `()`s. (it doesn't let you drop parens like in python in this case). `.requests` is a getter for `requests` on `getRandomizedRecipeRequests`. In Kotlin, it's highly encouraged to make variables `val`s instead of `var`s where a `val` is immutable and so it doesn't matter if other classes can access it because it can't be changed and in Java you would make getters and setters for it anyway. If you really don't want it to be visible external to the class, then you can mark it as private. I don't know if `.any()` is actually a function, but I'm sure there is a built in function that lets us pass in a Predicate to a function call on a `List` that returns whether or not there is anything that matches the predicate that might look something like this.

Reply:
Actually, I found [it](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/any.html)